contrib/src/main/java/com/datatorrent/contrib/avro/AvroRecordHelper.java

```diff
@@ -0,0 +1,124 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.contrib.avro;
+
+import java.text.ParseException;
+import java.util.List;
+
+import org.apache.avro.Schema;
+import org.apache.avro.Schema.Type;
+
+/**
+ * This is an utility class for Avro converted records
+ */
+
+public class AvroRecordHelper
+{
+
+  /**
+   * Convert a passed String value to the given type for the key as per Schema
+   * 
+   * @param schema
+   * @param key
+   * @param value
+   * @return
+   */
+
+  public static Object convertValueStringToAvroKeyType(Schema schema, String key, String value) throws ParseException
+  {
+    Type type = null;
+
+    if (schema.getField(key) != null) {
+      type = schema.getField(key).schema().getType();
+    } else {
+      return value;
+    }
+
+    Object convertedValue = null;
+
+    if (type == Type.UNION) {
+      convertedValue = convertAndResolveUnionToPrimitive(schema, key, value);
+    } else {
+      convertedValue = convertValueToAvroPrimitive(type, key, value);
+    }
+
+    return convertedValue;
+
+  }
+
+  private static Object convertValueToAvroPrimitive(Type type, String key, String value) throws ParseException
+  {
+    Object newValue = value;
+    switch (type) {
+      case BOOLEAN:
+        newValue = Boolean.parseBoolean(value);
+        break;
+      case DOUBLE:
+        newValue = Double.parseDouble(value);
+        break;
+      case FLOAT:
+        newValue = Float.parseFloat(value);
+        break;
+      case INT:
+        newValue = Integer.parseInt(value);
+        break;
+      case LONG:
+        newValue = Long.parseLong(value);
+        break;
+      case BYTES:
+        newValue = value.getBytes();
+        break;
+      case STRING:
+        newValue = value;
+        break;
+      case NULL:
+        newValue = null;
+        break;
+      default:
+        newValue = value;
+    }
+    return newValue;
+  }
+
+  private static Object convertAndResolveUnionToPrimitive(Schema schema, String key, String value) throws ParseException
+  {
+    Schema unionSchema = schema.getField(key).schema();
+    List<Schema> types = unionSchema.getTypes();
+    Object convertedValue = null;
+    for (int i = 0; i < types.size(); i++) {
+      try {
+        if (types.get(i).getType() == Type.NULL) {
+          if (value == null || value.equals("null")) {
+            convertedValue = null;
+            break;
+          } else {
+            continue;
+          }
+        }
+        convertedValue = convertValueToAvroPrimitive(types.get(i).getType(), key, value);
```

Comment:
Can you directly return from here and skip return on line 122? code would be more readable.